src/org/rascalmpl/uri/file/MavenRepositoryURIResolver.java

```diff
@@ -200,12 +200,18 @@ public String[] list(ISourceLocation uri) throws IOException {
      * @throws IOException if the root of the repo is gone or not permitted
      */
     private String[] listAllMainArtifacts() throws IOException {
-        try (Stream<Path> files = Files.walk(Paths.get(root.getPath()))) {
+        try (Stream<Path> files = Files.walk(Path.of(root.getURI()))) {
             return files.filter(f -> FileNameUtils.getExtension(f).equals("jar"))
             .filter(f -> !f.toString().endsWith("-sources.jar"))
             .filter(f -> !f.toString().endsWith("-javadoc.jar"))
             .map(f -> {
-                var fl = URIUtil.correctLocation("file", "", f.toString());
```

Comment:
this only works on linux/osx. always use `URIUtil` ;)

Reply:
ah yes, I mean, use `URIUtil.createFileLocation` and not `URIUtil.correctLocation` as that doesn't deal with path encoding & normalisation.

Reply:
good catch. This one could also be a nice one for a java static analysis linter specific for us.